rmgpy/data/surface.py

```diff
@@ -0,0 +1,282 @@
+
+        self.entries[label] = Entry(
+            index=index,
+            label=label,
+            metal=metal,
+            facet=facet,
+            surface_site_density=surfaceSiteDensity,
+            binding_energies=bindingEnergies or dict(),
+            short_desc=shortDesc,
+            long_desc=longDesc.strip(),
+        )
+
+    def load(self, path):
+        """
+        Load the metal library from the given path
+        """
+        Database.load(self, path, local_context={}, global_context={})
+
+    def save_entry(self, f, entry):
+        """
+        Write the given `entry` in the metal database to the file object `f`.
+        """
+        return save_entry(f, entry)
+
+    def get_binding_energies(self, label):
+        """
+        Get a metal's binding energies from its label
+        """
+        try:
+            return self.entries[label].binding_energies
+        except KeyError:
+            raise DatabaseError(f'Metal {label!r} not found in metal library database.')
+
+    def get_surface_site_density(self, label):
+        """
+        Get a metal's surface site desnity from its label
+        """
+        try:
+            return self.entries[label].surface_site_density
+        except KeyError:
+            raise DatabaseError(f'Metal {label!r} not found in metal library database.')
+
+    def get_all_entries_on_metal(self, metal_name):
+        """
+        Get all entries from the database that are on a certain metal, for any facet,
+        returning the labels.
+        """
+        matches = []
+        for label, entry in self.entries.items():
+            if entry.metal == metal_name:
+                matches.append(label)
+
+        if len(matches) is 0:
+            raise DatabaseError('Metal {0!r} not found in database'.format(metal_name))
+
+        return matches
+
+
+################################################################################
+
+class MetalDatabase(object):
+    """
+    A class for working with the RMG metal database.
+    """
+
+    def __init__(self):
+        self.libraries = {}
+        self.libraries['surface'] = MetalLibrary()
+        self.groups = {}
+        self.local_context = {}
+        self.global_context = {}
+
+    def __reduce__(self):
+        """
+        A helper function used when pickling a MetalDatabase object.
+        """
+        d = {
+            'libraries': self.libraries,
+        }
+        return (MetalDatabase, (), d)
+
+    def __setstate__(self, d):
+        """
+        A helper function used when unpickling a MetalDatabase object.
+        """
+        self.libraries = d['libraries']
+
+    def load(self, path, libraries=None, depository=True):
+        """
+        Load the metal database from the given `path` on disk, where `path`
+        points to the top-level folder of the solvation database.
+        
+        Load the metal library
+        """
+        self.libraries['surface'].load(os.path.join(path, 'libraries', 'metal.py'))
```

Comment:
Weird that the file is called `metal.py` but the key in the `libraries` dictionary is `surface`. Could they be consistent? Call it  `libraries['metal']` ?

Reply:
Right now, in the database it's in the `surface` folder and the library is called `metal.py`.  I was leaving it like this purposefully, so in the future we could have like a zeolite library, or a bimetallic library, or whatnot. But I'm also equally happy changing it.